src/doom/d_main.c

```diff
@@ -237,20 +266,43 @@ boolean D_Display (void)
 	D_PageDrawer ();
 	break;
     }
-    
+
     // draw buffered stuff to screen
     I_UpdateNoBlit ();
-    
+
     // draw the view directly
     if (gamestate == GS_LEVEL && (!automapactive || crispy->automapoverlay) && gametic)
     {
 	R_RenderPlayerView (&players[displayplayer]);
 
         // [crispy] Crispy HUD
-        if (screenblocks >= CRISPY_HUD)
+        if (screenblocks >= CRISPY_HUD || crispy->snowflakes)
             ST_Drawer(false, true);
     }
 
+    // [crispy] draw snowflakes
+    if(crispy->snowflakes)
+    {
+	int color = I_GetPaletteIndex(0xFF, 0xFF, 0xFF);
```

Comment:
Only once, please. Declare it as `static`, initialize with `-1` and only recalculate if it's `-1`.